flocker/node/agents/test/test_blockdevice.py

```diff
@@ -844,6 +844,15 @@ class IBlockDeviceAPITestsMixin(object):
     """
     Tests to perform on ``IBlockDeviceAPI`` providers.
     """
+    def tearDown(self):
+        """
+        Destroy volumes created for the cluster used by the test.
+        """
+        volumes = self.api.list_volumes()
+
+        for volume in volumes:
+            self.api.destroy_volume(volume.blockdevice_id)
```

Comment:
I think this approach breaks the LoopBackBlockDeviceAPI tests which have a separate cleanup function that removes all the backing files before this gets run...so the destroy_volume fails.

I don't know. I thought the aim was to be able to run tests for `create_volume` in isolation...without exercising `destroy_volume`...which is why I made `TidyCinderClient` to cleanup test volumes at a lower level.

One advantage of doing it that way is that we can measure the coverage of the just the `create_volume` related tests.

But @itamarst talked about this sort of thing when he was in Bristol last month....that our tests should exercise only our public APIs and supply realistic inputs and outputs...so I'd be interested to know what he thinks about this.